### ISSUE TITLE
CDATA sections with embedded XML are ignored

### ISSUE THREAD

**Issue by wspringer** [chrisbottin/xml-parser#4]
This document:

```xml
<foo><![CDATA[<baz/>]]></foo>
```

is parsed into this data structure:

```json
{
  "declaration": null,
  "root": {
    "type": "Element",
    "name": "foo",
    "attributes": {},
    "children": []
  },
  "children": [
    {
      "type": "Element",
      "name": "foo",
      "attributes": {},
      "children": []
    }
  ]
}
```

Notice that the CDATA section is excluded. If you change it to this:

```xml
<foo><![CDATA[baz]]></foo>
```

then the CDATA section *is* seen:

```json
{
  "declaration": null,
  "root": {
    "type": "Element",
    "name": "foo",
    "attributes": {},
    "children": [
      {
        "type": "CDATA",
        "content": "<![CDATA[baz]]>"
      }
    ]
  },
  "children": [
    {
      "type": "Element",
      "name": "foo",
      "attributes": {},
      "children": [
        {
          "type": "CDATA",
          "content": "<![CDATA[baz]]>"
        }
      ]
    }
  ]
}
```

Similarly, escaping the XML will include the data properly:

```xml
<foo>&lt;baz/&gt;</foo>
```

gives:

```json
{
  "declaration": null,
  "root": {
    "type": "Element",
    "name": "foo",
    "attributes": {},
    "children": [
      {
        "type": "Text",
        "content": "&lt;baz/&gt;"
      }
    ]
  },
  "children": [
    {
      "type": "Element",
      "name": "foo",
      "attributes": {},
      "children": [
        {
          "type": "Text",
          "content": "&lt;baz/&gt;"
        }
      ]
    }
  ]
}
```

So, what I would have expected in case of a document like this:

```xml
```

is this:

```json
{
  "declaration": null,
  "root": {
    "type": "Element",
    "name": "foo",
    "attributes": {},
    "children": [
      {
        "type": "CDATA",
        "content": "<![CDATA[<baz/>]]>"
      }
    ]
  },
  "children": [
    {
      "type": "Element",
      "name": "foo",
      "attributes": {},
      "children": [
        {
          "type": "CDATA",
          "content": "<![CDATA[<baz/>]]>"
        }
      ]
    }
  ]
}
```

**Comment by chrisbottin** [comment]
Thanks @wspringer for your comment, it's an odd one. I will look into it this week.

**Comment by chrisbottin** [comment]
Issue fixed in version 3.1.1
